src/server/scripts/Kalimdor/Firelands/boss_ragnaros_firelands.cpp

```diff
@@ -0,0 +1,5613 @@
+
+                me->CastSpell(me, SPELL_DREADFLAME_SPREAD_CONTROL_AURA, true);
+                me->CastSpell(me, SPELL_DREADFLAME_DELUGE_CONTROL_AURA, true);
+                me->CastSpell(me, SPELL_DREADFLAME_SPAWN_CONTROL_AURA, true);
+                me->CastSpell(me, SPELL_DREADFLAME_DAMAGE_CONTROL_AURA, true);
+            }
+
+            void JustRegisteredDynObject(DynamicObject* dynObj) override
+            {
+                RegisterDreadflame(dynObj);
+            }
+
+            void JustUnregisteredDynObject(DynamicObject* dynObj) override
+            {
+                UnregisterDreadflame(dynObj);
+            }
+
+            void RegisterDreadflame(DynamicObject* dynObj)
+            {
+                for (int i = 0; i < DreadFlamesMax; i++)
+                {
+                    if (std::sqrt(dynObj->GetExactDist2dSq(Dreadflames[i]->Position.GetPositionX(), Dreadflames[i]->Position.GetPositionY()) < 3.0f))
+                    {
+                        Dreadflames[i]->Owner = dynObj->GetGUID();
+                        break;
+                    }
+                }
+            }
+
+            void UnregisterDreadflame(DynamicObject* dynObj)
+            {
+                for (int i = 0; i < DreadFlamesMax; i++)
+                {
+                    if (Dreadflames[i]->Owner == dynObj->GetGUID())
+                    {
+                        Dreadflames[i]->Owner = ObjectGuid::Empty;
+                        break;
+                    }
+                }
+            }
+
+            std::list<Position> GetDreadflameSpreadPositions() const
+            {
+                std::vector<DreadflameInfo*> litDreadflames;
+                for (auto dreadflame : Dreadflames)
+                    if (!dreadflame->Owner.IsEmpty())
+                        litDreadflames.push_back(dreadflame);
+
+                std::list<Position> spreadPositions;
+                for (auto dreadflame : litDreadflames)
+                {
+                    std::list<DreadflameInfo*> unlitNearbyDreadflames;
+                    for (auto nearbyDreadflame : dreadflame->NearbyDreadflames)
+                    {
+                        if (nearbyDreadflame->Owner.IsEmpty())
+                            unlitNearbyDreadflames.push_back(nearbyDreadflame);
+                    }
+
+                    DynamicObject* fire = ObjectAccessor::GetDynamicObject(*me, dreadflame->Owner);
+                    std::list<Player*> players;
+                    Trinity::AnyPlayerInObjectRangeCheck checker(fire, 4.5f);
+                    Trinity::PlayerListSearcher<Trinity::AnyPlayerInObjectRangeCheck> searcher1(fire, players, checker);
+                    Cell::VisitWorldObjects(me, searcher1, 4.5f);
+                    players.remove_if(Trinity::UnitAuraCheck(false, SPELL_DELUGE_AURA));
+                    unlitNearbyDreadflames.remove_if([players](DreadflameInfo* dreadflame)
+                    {
+                        for
```

Comment:
i would prefer conditions here, all spells with *_ENTRY targets use conditions

Reply:
github on drugs, comment should be on https://github.com/TrinityCore/TrinityCore/pull/18058/files#diff-51e46c288e35e2e23cc48e781393423eR3859

Reply:
Yep, agreed.